File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/PostEnvironmentExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class PostEnvironmentExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey = new Dictionary<string, string> {["ApiKey"] = "YOUR_API_KEY"};

        var environmentPost = new EnvironmentPost(
            name: "My Environment",
            key: "environment-key-123abc",
            color: "DADBEE"
        );

        try
        {
            var response = new EnvironmentsApi(config).PostEnvironment(
                projectKey: null,
                environmentPost: environmentPost
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling EnvironmentsApi#PostEnvironment: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/PostEnvironmentExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const environmentPost: api.EnvironmentPost = {
  name: "My Environment",
  key: "environment-key-123abc",
  color: "DADBEE",
};

new api.EnvironmentsApi(configuration).postEnvironment(
  "projectKey_string", // projectKey
  environmentPost,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling EnvironmentsApi#postEnvironment:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
